src/unix/poll.c

```diff
@@ -32,9 +32,8 @@ static void uv__poll_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
   int pevents;
 
   handle = container_of(w, uv_poll_t, io_watcher);
-
-  if (events & POLLERR) {
-    uv__io_stop(loop, w, POLLIN | POLLOUT | UV__POLLRDHUP);
+  if ((events & POLLERR) && (events & UV__POLLPRI) != UV__POLLPRI) {
```

Comment:
That's what you might think, but i don't know why it's there. In my implementation it was the last line i added for it to work. Somehow the error is there but i don't know why or where i came from.